File: WrappersNET/MathLibNETTest/TestMul.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace MathLibNETTest
{
    [TestClass]
    public class TestMul
    {
        [TestMethod]
        [TestCategory("Mul")]
        [TestCategory("MathLib")]
        public void TestMultiplication1()
        {
            double val = MathLibNET.Math.Mul(1, 2);
            Assert.AreEqual(val, 2);
        }

        [TestMethod]
        [TestCategory("Mul")]
        [TestCategory("MathLib")]
        public void TestMultiplication2()
        {
            double val = MathLibNET.Math.Mul(2, 5);
            Assert.AreEqual(val, 10);
        }

        [TestMethod]
        [TestCategory("Mul")]
        [TestCategory("MathLib")]
        public void TestMultiplication3()
        {
            double val = MathLibNET.Math.Mul(-2, 5);
            Assert.AreEqual(val, -10);
        }

        [TestMethod]
        [TestCategory("Mul")]
        [TestCategory("MathLib")]
        public void TestMultiplication4()
        {
            double val = MathLibNET.Math.Mul(-2, -5);
            Assert.AreEqual(val, 10);
        }

        [TestMethod]
        [TestCategory("Mul")]
        [TestCategory("MathLib")]
        public void TestMultiplication5()
        {
            double val = MathLibNET.Math.Mul(0, -5);
            Assert.AreEqual(val, 0);
        }

        [TestMethod]
        [TestCategory("Mul")]
        [TestCategory("MathLib")]
        public void TestMultiplication6()
        {
            double val = MathLibNET.Math.Mul(5, 0);
            Assert.AreEqual(val, 0);
        }

        [TestMethod]
        [TestCategory("Mul")]
        [TestCategory("MathLib")]
        public void TestMultiplication7()
        {
            double val = MathLibNET.Math.Mul(0, 0);
            Assert.AreEqual(val, 0);
        }
    }
}

File: Native/MathLibTest/TestMul.cpp
#include "stdafx.h"
#include "CppUnitTest.h"
#include "MathLib.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace MathLibTest
{
    TEST_CLASS(TestMul)
    {
    public:

        BEGIN_TEST_METHOD_ATTRIBUTE(TestMul1)
            TEST_METHOD_ATTRIBUTE(L"Category", L"MathLib")
            TEST_METHOD_ATTRIBUTE(L"Category", L"Mul")
        END_TEST_METHOD_ATTRIBUTE()
        TEST_METHOD(TestMul1)
        {
            double d = Math::Mul(0, 10);
            Assert::AreEqual(0.0, d);
        }

        BEGIN_TEST_METHOD_ATTRIBUTE(TestMul2)
            TEST_METHOD_ATTRIBUTE(L"Category", L"MathLib")
            TEST_METHOD_ATTRIBUTE(L"Category", L"Mul")
        END_TEST_METHOD_ATTRIBUTE()
        TEST_METHOD(TestMul2)
        {
            double d = Math::Mul(10, 0);
            Assert::AreEqual(0.0, d);
        }

        BEGIN_TEST_METHOD_ATTRIBUTE(TestMul3)
            TEST_METHOD_ATTRIBUTE(L"Category", L"MathLib")
            TEST_METHOD_ATTRIBUTE(L"Category", L"Mul")
        END_TEST_METHOD_ATTRIBUTE()
        TEST_METHOD(TestMul3)
        {
            double d = Math::Mul(-5, -2);
            Assert::AreEqual(10.0, d);
        }
    };
}